docs/development/go/websocket/index.md

```diff
@@ -0,0 +1,651 @@
+
+The `for` loop at the end of the `main()` function is what orchestrates the program and deals with the channels, user input and UNIX interrupt handling.
+
+Have in mind that the `syscall.Kill(syscall.Getpid(), syscall.SIGINT)` statement sends the interrupt signal to the program using Go code. According to the logic of `WSclient.go`, the interrupt signal will make the program to close the WebSocket connection with the server and terminate its execution.
+
+#### Using the WebSocket client
+
+In this section you will learn how to use the `WSclient.go` WebSocket client. The `WSclient.go` utility is executed using the following format:
+
+    go run WSclient.go localhost:1234 /ws
+
+First, you define the host and the port number of the server without using `ws://` and then you give the endpoint you want to connect to.
+
+An interaction with the server that timeouts will look similar to the following:
+
+{{< output >}}
+Connecting to: localhost:1234 at /ws
+1
+2020/07/07 18:03:41 Received: 1
+2
+2020/07/07 18:03:41 Received: 2
+3
+2020/07/07 18:03:42 Received: 3
+4
+2020/07/07 18:03:42 Received: 4
+2020/07/07 18:03:46 Please give me input! 0
+2020/07/07 18:03:50 Please give me input! 1
+2020/07/07 18:03:54 Please give me input! 2
+2020/07/07 18:03:58 Please give me input! 3
+2020/07/07 18:04:02 Please give me input! 4
+2020/07/07 18:04:06 Please give me input! 5
+2020/07/07 18:04:06 Caught interrupt signal - quitting!
+2020/07/07 18:04:06 ReadMessage() error: websocket: close 1000 (normal)
+{{< /output >}}
+
+The output generated by the WebSocket server for the aforementioned interaction will be the following:
+
+{{< output >}}
+2020/07/07 18:03:25 Listening to TCP Port :1234
+2020/07/07 18:03:40 Connection from: localhost:1234
+2020/07/07 18:03:41 Received: 1
+2020/07/07 18:03:41 Received: 2
+2020/07/07 18:03:42 Received: 3
+2020/07/07 18:03:42 Received: 4
+2020/07/07 18:04:06 From localhost:1234 read websocket: close 1000 (normal)
+{{< /output >}}
+
+### Using `golang.org/x/net/websocket`
+
+The `golang.org/x/net/websocket` package offers another way of developing WebSocket clients and servers. This section will showcase how to implement a WebSocket client with it. The presented command line utility will read user input, send it to the WebSocket server and read the response from the WebSocket server before automatically quitting.
+
+As we are going to use the `golang.org/x/net/websocket` Go package in this subsection, it would be good to download it on your local machine by executing the next command:
+
+    go get -u golang.org/x/net/websocket
+
+{{< note >}}
+According to its documentation, `golang.org/x/net/websocket` lacks some features and you are advised to use `https://godoc.org/github.com/gorilla/websocket` or `https://godoc.org/nhooyr.io/websocket` instead. However, it is good to know about it as it might become part of the standard Go library if it ever gets properly updated.
```

Comment:
You should follow the advise in the documentation.  Do not use the golang.org/x/net/websocket package.  See https://github.com/golang/go/issues/18152 for more background.

Reply:
It just showcases how you can use golang.org/x/net/websocket package.

Reply:
Why would you showcase a package when the owners of the package discourage the package's use?

Reply:
The x/net/websocket package API is flawed.  The API will require substantial reversion if it's ever to become part of the standard library. 

The x/net/weboscket package does does correctly implement the websocket protocol.

Learning about the x/net/websocket package in the package's current state is a waste of time.

Reply:
Thank you for all your useful comments.